### ISSUE TITLE
Directories being <collect> are not following rules as expected

### ISSUE THREAD

**Issue by jtkerich** [ctron/rpm-builder#68]
I am collecting all the configuration files under a 2 directory trees to make the rpm.   Under rules I have a directory rule to do a chmod, user, and group.  However, none of the directories are set right and they are all owned by root and defaulting to 755.  If I use the include 1 directory example the target directory is created with the correct user/group and permission so that's working.  The example posted in the documentation does a chmod on any directory to 750, so I attempted to do only that but with 0770 but again the directories were all 755.   This is my pom.ml file.  I am doing a mvn package command on the command line via a java application on RH 8.6 using:

mvn -version
Apache Maven 3.8.6 (84538c9988a25aec085021c365c560670ad80f63)
Maven home: /usr/local/apache-maven-3.8.6
Java version: 1.8.0_242, vendor: Oracle Corporation, runtime: /usr/lib/jvm/java-1.8.0-openjdk-1.8.0.242.b08-1.el7.x86_64/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "3.10.0-1127.el7.x86_64", arch: "amd64", family: "unix"

example of what I am see after rpm -i the file:
drwxr-xr-x  4 root      root      4096 Nov 23 19:50 RTConfigurationFiles
drwxr-xr-x 8 root      root      4096 Nov 23 19:50 RTConfigurationFiles/configuration
-rw-rw---- 1 developer developer 1369 Nov 23 19:50 RTConfigurationFiles/metadata.xml
drwxr-xr-x 5 root      root      4096 Nov 23 19:50 RTConfigurationFiles/resources
etc.

[pom.xml.txt](https://github.com/ctron/rpm-builder/files/10078901/pom.xml.txt)

The pom.xml is successfully setting all the files to the user account and the target path is to the user home directory as expected.  The only thing that is wrong is every directory is root root 755 vs. developer developer 770.

john.kerich@frequentisdefense.com

P.S.
I also emailed if it was possible to directly call the lib classes from a java application to make the rpm via the application running mvn on the command line to do it.

**Comment by ctron** [comment]
I looks ok to me. Then again, I do recall rules always being a bit complex to understand (for myself). One way to deal with this, is to explicitly configure things like owners and permissions, which can be done as part of the entry. True, that limits the use of collectors.

You can also turn on Maven logging (`mvn -X`), which should print out all kinds of information on evaluating the rules, and might shed some light on what is happening.

As to the question if RPM builder can be used as a library: absolutely, yes. The Maven plugin is just using the Eclipse Packager project's ability to create RPM files: https://github.com/eclipse/packager/tree/master/rpm and https://www.javadoc.io/doc/org.eclipse.packager/packager-rpm/latest/org/eclipse/packager/rpm/build/package-summary.html

Maybe take a look at the [WriterTest.java](https://github.com/eclipse/packager/blob/master/rpm/src/test/java/org/eclipse/packager/rpm/WriterTest.java), which makes use of this.

**Comment by jtkerich** [comment]
I updated the pom.xml to remove the env variables and hardcode the user/group and path and this time it worked.  I then restored the original pom.xml and tried that again and it worked.   So, I am not sure what was going on, but it's now working as expected on two different Linux versions. The only other difference is the file uploaded is dos while I am Linux.  I ran dos2unix after I posted, but I don't see how that would cause a problem.  Its working now.
